src/runtime-rs/crates/persist/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+pub mod hypervisor;
+pub mod network;
+pub mod sandbox;
+use anyhow::{anyhow, Context, Ok, Result};
+use serde::de;
+use std::{fs::File, io::BufReader};
+pub const KATA_PATH: &str = "/run/kata";
+pub const PERSIST_FILE: &str = "state.json";
+use async_trait::async_trait;
+use safe_path::scoped_join;
+#[async_trait]
+pub trait Persist
+where
+    Self: Sized,
+{
+    /// The type of the object representing the state of the component.
+    type State;
+    /// The type of the object holding the constructor arguments.
+    type ConstructorArgs;
+
+    /// Returns the state of the component.
+    async fn save(&self) -> Result<Self::State>;
+    /// Constructs a component from a specified state.
+    async fn restore(constructor_args: Self::ConstructorArgs, state: &Self::State) -> Result<Self>;
+}
+
+pub fn to_disk<T: serde::Serialize>(value: &T, sid: &str) -> Result<()> {
+    let mut path = scoped_join(KATA_PATH, sid)?;
```

Comment:
Thanks - this ensures `sid` is below `KATA_PATH`. However, the code is still not validating `sid` to make sure it doesn't contain characters like `.` and `/` which are not allowed so please can you add calls to [`verify_id()`](https://github.com/kata-containers/kata-containers/blob/runtime-rs/src/libs/kata-sys-util/src/validate.rs#L17) here and in `from_disk()` below.

Reply:
Thanks! Fixed it.